packages/handlersjs-core/lib/logging/logger.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { Logger } from './logger';
+import { LoggerLevel } from './logger-level';
+
+describe('Logger', () => {
+
+  class MockLogger extends Logger {
+
+    log(level: LoggerLevel, typeName: string, message: string, data?: unknown): void {
+
+      throw new Error('Method not implemented.');
+
+    }
+
+    constructor(minimumLevel: LoggerLevel, minimumLevelPrintData: LoggerLevel) {
+
+      super(minimumLevel, minimumLevelPrintData);
+
+    }
+
+  }
+
+  const logger = new MockLogger(LoggerLevel.info, LoggerLevel.info);
+
+  describe('info', () => {
```

Comment:
These tests only describe the parameter checks. What do we expect the logger to do?
Also: since all parameter checks are the same, you can abstract from them in a separate function and call it for each of the methods.

Reply:
> Also: since all parameter checks are the same, you can abstract from them in a separate function and call it for each of the methods.

You would still want the expect statement to expect the correct function to throw the error. If I moved this to a seperate function like `const paramCheck = (fn: string, typeName: string, message: string) => {
 if (!typeName) { return expect(() => logger.fn(typename, message).toThrow('Typename should be set');}
...
}` how will I call the correct logger function? cause logger.fn doesn't work and if I add a switch statement that checks the fn string, I'd end up with more lines of code than before.

Reply:
I was thinking about something like (you were close: instead of a string, just pass the function itself)
```
const paramCheck = (logFunction: (typeName: string, msg: string) => void) => {
  expect(() => logFunction(undefined, 'msg')).toThrow('Typename should be set');
  expect(() => logFunction(null, 'msg')).toThrow('Typename should be set');
  expect(() => logFunction('string', undefined)).toThrow('Message should be set');
  expect(() => logFunction('string', null)).toThrow('Message should be set');
}
```
And then for a specific function call `paramCheck(logger.warn)`, for example.

Reply:
> I was thinking about something like (you were close: instead of a string, just pass the function itself)
> 
> ```
> const paramCheck = (logFunction: (typeName: string, msg: string) => void) => {
>   expect(() => logFunction(undefined, 'msg')).toThrow('Typename should be set');
>   expect(() => logFunction(null, 'msg')).toThrow('Typename should be set');
>   expect(() => logFunction('string', undefined)).toThrow('Message should be set');
>   expect(() => logFunction('string', null)).toThrow('Message should be set');
> }
> ```
> 
> And then for a specific function call `paramCheck(logger.warn)`, for example.

Alright, didn't think of passing functions. Thanks.